pkg/bicep/schematobicep.go

```diff
@@ -61,3 +64,196 @@ func getBicepType(schemaType string) (string, error) {
+		joinList = append(joinList, d)
+	}
+	bicepObj := strings.Join(joinList, " ")
+	buf.WriteString(fmt.Sprintf("@allowed(%v)\n", bicepObj))
+}
+
+func declareDescription(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Description != "" {
+		// decorators are in sys namespace. to avoid potential collision with other parameters named "description", we use "sys.description" instead of just "description" https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#decorators
+		buf.WriteString(fmt.Sprintf("@sys.description('%s')\n", sch.Description))
+	}
+}
+
+func declareMinValue(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Minimum != "" {
+		buf.WriteString(fmt.Sprintf("@minValue(%v)\n", sch.Minimum))
+	}
+}
+
+func declareMaxValue(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Maximum != "" {
+		buf.WriteString(fmt.Sprintf("@maxValue(%v)\n", sch.Maximum))
+	}
+}
+
+func declareMinLength(sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	if bicepType == "array" {
+		if sch.MinItems != nil {
+			buf.WriteString(fmt.Sprintf("@minLength(%d)\n", *sch.MinItems))
+		}
+		return nil
+	}
+
+	if sch.MinLength != nil {
+		buf.WriteString(fmt.Sprintf("@minLength(%d)\n", *sch.MinLength))
+	}
+
+	return nil
+}
+
+func declareMaxLength(sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	if bicepType == "array" {
+		if sch.MaxItems != nil {
+			buf.WriteString(fmt.Sprintf("@maxLength(%d)\n", *sch.MaxItems))
+		}
+		return nil
+	}
+
+	if sch.MaxLength != nil {
+		buf.WriteString(fmt.Sprintf("@maxLength(%d)\n", *sch.MaxLength))
+	}
+
+	return nil
+}
+
+func declareSecure(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Format == "password" {
+		buf.WriteString("@secure()\n")
+	}
+}
+
+func declareDefault(name string, sch *schema.Schema, buf *bytes.Buffer) error {
+	if sch.Default != nil {
+		bicepType, err := getBicepType(sch.Type)
+		if err != nil {
+			return err
+		}
+
+		switch bicepType {
+		case "string":
+			buf.WriteString(fmt.Sprintf("param %s %s = '%s'\n", name, bicepType, sch.Default))
+		case "int":
+			buf.WriteString(fmt.Sprintf("param %s %s = %v\n", name, bicepType, sch.Default))
+		case "bool":
+			buf.WriteString(fmt.Sprintf("param %s %s = %t\n", name, bicepType, sch.Default))
+		case "array":
+			declareDefaultArray(name, sch, buf)
+		case "object":
+			declareDefaultObject(name, sch, buf)
+		}
+	}
+	return nil
+}
+
+func declareDefaultArray(name string, sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	defBytes, err := json.MarshalIndent(sch.Default.([]interface{}), "", "    ")
+	if err != nil {
+		return err
+	}
+
+	defString := string(defBytes)
+	r := strings.NewReplacer(`"`, `'`, ",", "")
+
+	buf.WriteString(fmt.Sprintf("param %s %s = %v\n", name, bicepType, r.Replace(defString)))
```

Comment:
What happens if this is an array of objects? There is code in `declareDefaultObject` specific to object handling, which won't get executed here.